src/publish-build/index.ts

```diff
@@ -0,0 +1,90 @@
+import * as core from '@actions/core';
+import {
+  S3Client,
+  PutObjectCommand,
+  PutObjectCommandOutput,
+  PutObjectCommandInput,
+} from '@aws-sdk/client-s3';
+import { promises as fs, createReadStream } from 'fs';
+import mime from 'mime';
+import * as path from 'path';
+
+const REQUIRED_ENV_VARS = [
+  'AWS_ACCESS_KEY_ID',
+  'AWS_SECRET_ACCESS_KEY',
+  'AWS_BUCKET',
+  'PROJECT',
+  'SOURCE_DIR',
+  'DESTINATION_DIR',
+  'COMMIT_HASH',
+];
+
+function validateEnvVars(): void {
```

Comment:
NIT: We could have this return an object of the validated env vars, that way we can access those values again without having to do `process.env.['VAR'] ?? ''` every time.

Reply:
done!